### PR TITLE
Make rules use comment iterators

### PR DESCRIPTION
**Pull request by TheSavior**
This is necessary for the changes we want in #1028

### CODE REVIEW

File: lib/rules/disallow-capitalized-comments.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ module.exports.prototype = {
         var letterPattern = require('../../patterns/L');
         var lowerCasePattern = require('../../patterns/Ll');
 
-        file.getComments().forEach(function(comment) {
+        file.iterateTokensByType(['Line', 'Block'], function(comment) {
```
Comment (TheSavior) [inline lib/rules/disallow-capitalized-comments.js]: It definitely feels weird to iterate by the types Line and Block and expect only comments. This should be renamed to LineComments and BlockComments but it appears that the code that is doing that is deeper than our project. At least, I couldn't find where those were being set in JSCS.

Comment (TheSavior) [inline lib/rules/disallow-capitalized-comments.js]: I think that a rename change to how these tokens are typed should be a separate standalone PR.

**Comment by TheSavior** [comment]
This actually has a bit more nuance to it.

@mdevils In js-file lines ~28 now has this._tree.tokens and this._tokens
```javascript
this._tree.tokens = this._fixEs6Tokens(this._tree.tokens);
this._tokens = this._buildTokenList(this._tree.tokens, this._tree.comments);
```

`getTokens` was returning only this._tree.tokens, but should probably now return this._tokens. Since we are now adding the EOF token as well, that means the tests for js-file need to be updated if we make this switch. It seems like the right switch to be making. Is there something I'm missing, or should I go ahead with this?

**Comment by mikesherov** [comment]
we should now be using `this._tokens` everywhere, right @mdevils? That was my intent at least... and yeah, `getComments` should be deprecated.

File: lib/js-file.js [hunk 2]
```diff
@@ -499,7 +499,7 @@ JsFile.prototype = {
      * @returns {Object[]}
      */
     getTokens: function() {
-        return this._tree.tokens;
```
Comment (mikesherov) [inline lib/js-file.js]: no, the better thing here is `this._tokens`.

**Comment by mikesherov** [comment]
Also, can you rebase this please?

File: lib/js-file.js (hunk 2 above)
Comment (zxqfox) [inline lib/js-file.js]: Yeah. `_tree.tokens.concat(_tree.comments)` ordered differently than `_tokens`.

**Comment by mdevils** [comment]
@mikesherov Yes, looks like `getTokens` should return combined token list (with comments).

File: lib/rules/disallow-capitalized-comments.js (hunk 1 above)
Comment (mdevils) [inline lib/rules/disallow-capitalized-comments.js]: Yeah, I thought so too... `LineComment` and `BlockComment` has clearer meaning.

Comment (zxqfox) [inline lib/rules/disallow-capitalized-comments.js]: :+1:

**Comment by TheSavior** [comment]
@mdevils. Combined being `this._tokens` or `this._tree.tokens.concat(this._tree.comments)`? It seems like it should be `this._tokens`

**Comment by zxqfox** [comment]
@TheSavior Difference just in order. `this._tokens` are better since the set is the same.

**Comment by mikesherov** [comment]
@TheSavior, yes, just `this._tokens`

**Comment by TheSavior** [comment]
There is actually one other difference, which is what breaks tests currently. `this._tokens` has EOF:

`this._tokens:`
```
['var', 'a', '// com1', 'var', 'b', '// com2', EOF]
```

`this._tree.tokens.concat(this._tree.comments):`

```
['var', 'a', 'var', 'b', '// com1', '// com2']
```

File: lib/rules/disallow-capitalized-comments.js (hunk 1 above)
Comment (TheSavior) [inline lib/rules/disallow-capitalized-comments.js]: How do we actually do this? I don't see where the types for those tokens are being defined.

Comment (zxqfox) [inline lib/rules/disallow-capitalized-comments.js]: As I see it comes from esprima code base so the only thing we can do is redefining.

Comment (TheSavior) [inline lib/rules/disallow-capitalized-comments.js]: It's weird, but lets just be consistent with esprima.

**Comment by TheSavior** [comment]
Alright, I think this is good for a final review.

File: lib/rules/disallow-capitalized-comments.js (hunk 1 above)
Comment (zxqfox) [inline lib/rules/disallow-capitalized-comments.js]: Actually, we don't need to. It's not a part of Parser API because comments doesn't affect code execution and they are just skipped on parsing for execution. Since we merging comment tokens into code ASD we already using superset of Parser API and `Line` and `Block` types are confusing. I suggest to fix that and to describe it in JSFile docs.

**Comment by mikesherov** [comment]
Thanks!
